test/runtests.jl

```diff
@@ -42,7 +42,7 @@ Logging.with_logger((islogging || Pkg.DEFAULT_IO[] == devnull) ? Logging.Console
         iob = IOBuffer()
         Pkg.activate(; temp = true)
         try
-            Pkg.add("HistoricalStdlibVersions", io=iob) # Needed for custom julia version resolve tests
+            Pkg.add(name="HistoricalStdlibVersions", version="1.2", uuid="6df8b67a-e8a0-4029-b4b7-ac196fe72102", io=iob) # Needed for custom julia version resolve tests
```

Comment:
Maybe we should parse the compat bound from the Project.toml?

Reply:
I thought about that, but what if we start doing `"1.2,2"`. Part of the problem is this code path isn't hit on Pkg's own CI.

We might just want to teach `Base.runtests` to run stdlib tests via `Pkg.test` to handle test deps properly

Reply:
Is `Pkg.test` guaranteed to pick up the correct version of the stdlib? (Because we're not doing the "modify the stdlib UUID trick.)

If we can get `Pkg.test("StdlibName")` to work correctly in `Base.runtests`, I agree that's probably the best solution.

Reply:
> I thought about that, but what if we start doing `"1.2,2"`.

That's exactly why I want this.  `Pkg.add(;..., version=VersionSpec(...))` works, so you can do something like the following:

```
hsv_compat_val = TOML.parsefile(project_path)["compat"]["HistoricalStdlibVersions"]
Pkg.add(;name="HistoricalStdlibVersions", version=Pkg.Types.semver_spec(hsv_compat_val))
```

Reply:
IIRC, we do some shenanigans with the `JULIA_PROJECT` and `JULIA_LOAD_PATH`. So we need to make sure that `Pkg.test` still works correctly in that case.

Also, how will this work with testgroups? For stdlibs, `Base.runtests` is supposed to look at the `test/testgroups` file, and run each testgroup in parallel. Would we have to teach `Pkg.test` to understand stdlib testgroups? I'm not sure if we want to add that kind of functionality as public API in Pkg, but maybe we can add it as a private/internal Pkg API that we only use for stdlib tests in `Base.runtests`.

Reply:
Something like `Pkg.test("StdlibName"; _stdlib_testgroup = "desired_test_group")` might be sufficient. And we keep `_stdlib_testgroup` as private/internal API only.